pypsa/optimization/optimize.py

```diff
@@ -330,7 +338,10 @@ def assign_solution(n):
             else:
                 set_from_frame(n, c, attr, df)
         else:
-            n.df(c)[attr + "_opt"].update(df)
+            if attr == "n_opt":
+                n.df(c)[attr] = df
+            else:
+                n.df(c)[attr + "_opt"].update(df)
```

Comment:
Let's keep the formulation as before and have the optimal number of components named "n_mod_opt" (n_mod + _opt).
To do so, we just need to keep the original formulation